### PR TITLE
Make epan_auto_reset_count configurable

### PR DESCRIPTION
**Pull request by assafrabin**

### CODE REVIEW

**Review by yehonatanz — CHANGES_REQUESTED** [review]
Just please accept `epan_auto_reset_count` as an argument to `__init__`

File: tests/marine/test_marine.py [hunk 1]
```diff
@@ -388,3 +388,13 @@ def test_filter_and_parse_with_no_parameters(
...
+
+
+def test_get_epan_auto_reset_count_return_default_value(marine_instance: Marine):
+    assert marine_instance.epan_auto_reset_count == 20000
+
+
+def test_set_epan_auto_reset_count_return_default_value(marine_instance: Marine):
+    SOME_VALUE = 1
+    marine_instance.epan_auto_reset_count = SOME_VALUE
+    assert marine_instance.epan_auto_reset_count == SOME_VALUE
```
Comment (tomlegkov) [inline tests/marine/test_marine.py:417]: I think the convention is `expected == actual`, but I'll mark it as approved and leave it up to @yehonatanz to decide

**Review by tomlegkov — APPROVED** [review]

File: marine/marine.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class MarineResult(Structure):
 
 
 class Marine:
-    def __init__(self, lib_path):
+    def __init__(self, lib_path, epan_auto_reset_count=None):
```
Comment (tomlegkov) [inline marine/marine.py]: Add typing

**Review by tomlegkov — APPROVED** [review]
Tiny change and it looks good :)

File: tests/fixtures/marine/marine_fixtures.py [hunk 3]
```diff
@@ -25,6 +25,15 @@ def tcp_packet() -> bytes:
...
 @pytest.fixture(scope="session")
-def marine_instance() -> Marine:
-    path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "libmarine.so")
-    return Marine(path)
+def libmarine_path():
+    return os.path.join(os.path.dirname(os.path.abspath(__file__)), "libmarine.so")
+
+
+@pytest.fixture(scope="session")
+def epan_auto_reset_count():
```
Comment (yehonatanz) [inline tests/fixtures/marine/marine_fixtures.py]: `-> int`

File: tests/fixtures/marine/marine_fixtures.py [hunk 4]
```diff
@@ -25,6 +25,15 @@ def tcp_packet() -> bytes:
 
 
 @pytest.fixture(scope="session")
-def marine_instance() -> Marine:
-    path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "libmarine.so")
-    return Marine(path)
+def libmarine_path():
```
Comment (yehonatanz) [inline tests/fixtures/marine/marine_fixtures.py]: `-> str`

File: tests/fixtures/marine/marine_fixtures.py [hunk 5]
```diff
@@ -25,6 +25,15 @@ def tcp_packet() -> bytes:
...
+    return os.path.join(os.path.dirname(os.path.abspath(__file__)), "libmarine.so")
+
+
+@pytest.fixture(scope="session")
+def epan_auto_reset_count():
+    return 5000
+
+
+@pytest.fixture(scope="session")
+def marine_instance(libmarine_path, epan_auto_reset_count) -> Marine:
```
Comment (yehonatanz) [inline tests/fixtures/marine/marine_fixtures.py]: typing

**Review by yehonatanz — CHANGES_REQUESTED** [review]
I liked the `setter`/`getter`, didn't mean for you to remove them.

File: tests/fixtures/marine/marine_fixtures.py [hunk 6]
```diff
@@ -25,6 +25,15 @@ def tcp_packet() -> bytes:
...
+    return os.path.join(os.path.dirname(os.path.abspath(__file__)), "libmarine.so")
+
+
+@pytest.fixture(scope="session")
+def epan_auto_reset_count() -> int:
+    return 5000
+
+
+@pytest.fixture(scope="session")
+def marine_instance(libmarine_path: str, epan_auto_reset_count: int) -> object:
```
Comment (yehonatanz) [inline tests/fixtures/marine/marine_fixtures.py]: `-> Marine`

File: marine/marine.py [hunk 7]
```diff
@@ -13,7 +13,7 @@ class MarineResult(Structure):
 
 
 class Marine:
-    def __init__(self, lib_path):
+    def __init__(self, lib_path: str, epan_auto_reset_count: int = None):
```
Comment (yehonatanz) [inline marine/marine.py]: `Optional[int]`

**Review by yehonatanz — CHANGES_REQUESTED** [review]
Tiny typing notes.
But also, I'd like to see back the `setter/getter`

**Review by yehonatanz — APPROVED** [review]

**Pull request merged**